### PR TITLE
refactor: upgrade to deno_ast 0.33

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: examples/dlint/color.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
```
Comment (dsherret) [inline examples/dlint/color.rs:2]: This space is necessary to prevent auto-imports being inserted above the copyright statement.

Comment (bartlomieju) [reply examples/dlint/color.rs:2]: Nice, I've hit it so many times...

File: examples/dlint/diagnostics.rs
Comment (bartlomieju) [inline examples/dlint/diagnostics.rs:1]: Lovely!

File: src/context.rs [hunk 2]
```diff
@@ -59,8 +60,8 @@ impl<'view> Context<'view> {
     }
   }
 
-  /// File name on which the lint rule is run
-  pub fn file_name(&self) -> &str {
+  /// File specifier on which the lint rule is run.
+  pub fn specifier(&self) -> &ModuleSpecifier {
```
Comment (bartlomieju) [inline src/context.rs:64]: 👍

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**